sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,143 @@
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        placeContent: 'space-between',
+        alignItems: 'center',
+        pl: 2,
+        pr: 1,
+        py: 1,
+        ...appBarProps?.sx
+      }}
+    >
+      {children}
+    </AppBar>
+  );
+}
 
 export interface ChatGPTPopoverProps extends PopoverProps {
   appBarTitle?: string;
   width?: number | string;
   height?: number | string;
   appBarProps?: AppBarProps;
-  chatGPTProps: ChatGPTProps;
+  chatGPTProps?: ChatGPTProps;
+  isMinimized?: boolean;
+  onMinimize?: () => void;
+  onMaximize?: () => void;
 }
 
-function ChatGPTPopover(props: ChatGPTPopoverProps) {
+function ChatGPTPopover(props: Readonly<ChatGPTPopoverProps>) {
   const {
     open,
     onClose,
     appBarProps,
     chatGPTProps,
+    isMinimized = false,
+    onMinimize,
+    onMaximize,
     appBarTitle = 'AI Assistant',
     width = 450,
     height = 500,
     ...popoverProps
   } = props;
+
+  const handleMinimize = useCallback(() => {
+    onMinimize?.();
+  }, [onMinimize]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !isMinimized) {
+        handleMinimize();
+      }
+    };
+
+    if (open) {
+      document.addEventListener('keydown', handleKeyDown);
+    }
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [open, isMinimized, handleMinimize]);
+
   return (
-    <Popover
-      open={open}
-      onClose={onClose}
-      keepMounted={false}
-      anchorReference="none"
-      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
-      anchorPosition={{ top: 100, left: 100 }}
-      BackdropProps={{ invisible: false, sx: { backgroundColor: 'rgba(0, 0, 0, 0.5)' } }}
-      {...popoverProps}
-      sx={{
-        [`> .${paperClasses.root}`]: {
-          width,
-          height,
-          display: 'flex',
-          flexDirection: 'column',
-          bottom: 10,
-          right: 10
-        },
-        ...popoverProps?.sx
-      }}
-    >
-      <AppBar
-        position="static"
-        {...appBarProps}
+    <>
+      <Popover
+        open={open}
+        onClose={onClose}
+        disableEscapeKeyDown={true}
+        keepMounted={false}
+        anchorReference="none"
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
+        anchorPosition={{ top: 100, left: 100 }}
+        BackdropProps={{
+          invisible: false,
+          sx: { backgroundColor: 'rgba(0, 0, 0, 0.5)' },
+          onClick: (event) => {
+            event.stopPropagation();
+            handleMinimize();
+          }
+        }}
+        {...popoverProps}
         sx={{
-          display: 'flex',
-          flexDirection: 'row',
-          placeContent: 'space-between',
-          alignItems: 'center',
-          pl: 2,
-          pr: 1,
-          py: 1,
-          ...appBarProps?.sx
+          visibility: isMinimized ? 'hidden' : 'visible',
```

Comment:
Using visibility can interfere with the pointer events of the area where this is hidden at. Should use the `open` prop. For example `open={open && !isMinimized}`.

Reply:
@rart Using `open={open && !isMinimized}` seems to close the component completely. This is a problem for the plugin since we want to keep the chat history when minimizing. 
Is there a better way other than using the visibility to keep the chat session when minimizing? I also tried with `opacity: 0` but this does not work well with the `Popover` component.

Reply:
You need to add `keepMounted` prop to the Popover.

Reply:
Perhaps try `keepMounted={isMinimized}` so that it doesn't stay mounted unnecessarily.

Reply:
Updated